R/summariseOmopTable.R

```diff
@@ -22,133 +22,106 @@ summariseOmopTable <- function(omopTable,
+    }
+    if(sourceVocabulary){
+      cli::cli_warn("sourceVocabulary turned to FALSE, as omopTable provided is observation_period")
+      sourceVocabulary <- FALSE
+    }
+    if(domainId){
+      cli::cli_warn("domainId turned to FALSE, as omopTable provided is observation_period")
+      domainId <- FALSE
+    }
   }
 
   cdm <- omopgenerics::cdmReference(omopTable)
   omopTable <- omopTable |> dplyr::ungroup()
 
-  # counts summary
-  cli::cli_inform("Summarising counts")
-  persons <- cdm[["person"]] |>
-    dplyr::ungroup() |>
-    dplyr::summarise("n" = as.integer(dplyr::n())) |>
-    dplyr::pull("n")
-  result <- omopTable |>
-    dplyr::summarise(
-      "number_records" = dplyr::n(),
-      "number_subjects" = dplyr::n_distinct(.data$person_id)
-    ) |>
-    dplyr::collect() |>
-    dplyr::mutate(dplyr::across(dplyr::everything(), as.integer)) |>
-    dplyr::mutate(
-      "subjects_percentage" = 100 * .data$number_subjects / .env$persons
-    ) |>
-    dplyr::mutate(dplyr::across(dplyr::everything(), as.character)) |>
-    tidyr::pivot_longer(
-      cols = dplyr::everything(),
-      names_to = "variable_name",
-      values_to = "estimate_value"
-    ) |>
-    dplyr::mutate(
-      "variable_level" = NA_character_,
-      "estimate_name" = dplyr::if_else(
-        grepl("number", .data$variable_name), "count", "percentage"
-      ),
-      "estimate_type" = dplyr::if_else(
-        grepl("number", .data$variable_name), "integer", "percentage"
-      )
-    )
-  den <- result |>
-    dplyr::filter(.data$variable_name == "number_records") |>
-    dplyr::pull("estimate_value") |>
-    as.integer()
-
-  # records per person
-  if (length(recordsPerPerson) > 0) {
-    cli::cli_inform("Summarising records per person")
-    suppressMessages(
-      result <- result |>
-        dplyr::union_all(
-          cdm[["person"]] |>
-            dplyr::left_join(
-              omopTable |>
-                dplyr::group_by(.data$person_id) |>
-                dplyr::summarise(
-                  "records_per_person" = as.integer(dplyr::n()),
-                  .groups = "drop"
-                ),
-              by = "person_id",
-            ) |>
-            dplyr::mutate("records_per_person" = dplyr::if_else(
-              is.na(.data$records_per_person),
-              0L,
-              .data$records_per_person
-            )) |>
-            PatientProfiles::summariseResult(
-              variables = "records_per_person",
-              estimates = recordsPerPerson,
-              counts = FALSE
-            ) |>
-            dplyr::select(
-              "variable_name", "variable_level", "estimate_name",
-              "estimate_type", "estimate_value"
-            )
-        )
-    )
+  people <- getNumberPeopleInCdm(cdm)
+  result <- omopgenerics::emptySummarisedResult()
+
+  if(omopTable |> dplyr::tally() |> dplyr::pull("n") == 0){
+    cli::cli_abort(paste0(omopgenerics::tableName(omopTable), " omop table is empty."))
```

Comment:
can we just throw a warning a return an empty result?
In principle the error is for when the inputs are not correctly populated